search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -0,0 +1,146 @@
+                  :let [f (target-file entry)]]
+            (debug (format "Zip file entry: [%s]" (.getName entry)))
+            (io/copy (.getInputStream zip entry) f))))
+      (.toFile target-dir))
+    (catch Exception e
+      (errors/throw-service-error :bad-request (str "Error while uncompressing zip file: " (.getMessage e))))))
+
+(defn find-shp-file
+  "Find the .shp file in the given directory (File) and return it as a File"
+  [dir]
+  (let [files (file-seq dir)]
+    (first (filter #(= "shp" (FilenameUtils/getExtension (.getAbsolutePath %))) files))))
+
+(defn geometry->conditions
+  "Get one or more conditions for the given Geometry. This will only
+  return more than one condition if the Geometry is a GeometryCollection."
+  [geometry]
+  (let [num-geometries (.getNumGeometries geometry)]
+    (for [index (range 0 num-geometries)
+          :let [sub-geometry (.getGeometryN geometry index)]]
+      (geo/geometry->condition sub-geometry))))
+
+(defn transform-to-epsg-4326
+  "Transform the geometry to WGS84 CRS if is not already"
+  [geometry src-crs]
+  ;; if the source CRS is defined and not already WGS84 then transform the geometry to WGS84
+  (if (and src-crs
+           (not (= (.getName src-crs) (.getName EPSG-4326-CRS))))
+    (let [src-crs-name (.getName src-crs)
+          _ (debug (format "Source CRS: [%s]" src-crs-name))
+          _ (debug (format "Source axis order: [%s]" (CRS/getAxisOrder src-crs)))
+          _ (debug (format "Destination CRS: [%s]" (.getName EPSG-4326-CRS)))
+          _ (debug (format "Destination axis order: [%s]" (CRS/getAxisOrder EPSG-4326-CRS)))
+          transform (CRS/findMathTransform src-crs EPSG-4326-CRS false)]
+      ; if we didn't find a tranform send an error message
+      (if transform
+        (let [new-geometry (JTS/transform geometry transform)
+              _ (debug (format "New geometry: [%s" new-geometry))]
+          new-geometry)
+        (errors/throw-service-error :bad-request (format "Cannot transform source CRS [%s] to WGS 84" src-crs-name))))
+    geometry))
+
+(defn feature->conditions
+  "Process the contents of a Feature to return query conditions"
+  [feature]
+  (let [crs (when (.getDefaultGeometryProperty feature)
+              (-> feature .getDefaultGeometryProperty .getDescriptor .getCoordinateReferenceSystem))
+        properties (.getProperties feature)
+        _ (doseq [p properties] (debug (.getName p)))
+        _ (doseq [p properties] (debug (.getValue p)))
+        geometry-props (filter (fn [p] (geo/geometry? (.getValue p))) properties)
+        _ (debug (format "Found [%d] geometries" (count geometry-props)))
+        geometries (map #(-> % .getValue (transform-to-epsg-4326 crs)) geometry-props)
+        _ (debug (format "Transformed [%d] geometries" (count geometries)))]
+    (flatten (map (fn [g] (geometry->conditions g)) geometries))))
+
+(defn esri-shapefile->condition-vec
```

Comment:
Add unit tests for this function that exercise each of the different exception cases.

Reply:
Done.